src/app/code/community/Shopgate/Cloudapi/Helper/Api2/Quote.php

```diff
@@ -134,16 +135,27 @@ function (Mage_Sales_Model_Quote_Item $item) {
      */
     public function addTotals(Mage_Sales_Model_Quote $quote)
     {
-        $totals = $quote->getTotals();
+        $totals = array_map(
+            function (Mage_Sales_Model_Quote_Address_Total $total) {
+                return $total->getData();
+            },
+            $quote->getTotals()
+        );
+
+        if (isset($totals[self::KEY_SHIPPING]) && $totals[self::KEY_SHIPPING]['value']) {
+            $totals[self::KEY_SHIPPING]['value'] = Mage::helper('tax')->getShippingPrice(
```

Comment:
@andre-kraus just a general safety concern on this line, it would to check the value's existence as well:
```
if (isset($totals['test'], $totals['test']['value'])) {}
```
As it will fail in the scenario where shipping is set, but `value` is not set. I am, of course, not sure if it is possible in vanilla mage, but when dealing with collected totals, everything is possible as they are being overwritten.